src/mpid/ch4/shm/posix/posix_coll_release_gather.h

```diff
@@ -125,13 +128,13 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_mpi_bcast_release_gather(void *buffer,
     MPL_DBG_MSG_FMT(MPIR_DBG_COLL, VERBOSE, (MPL_DBG_FDEST, "Bcast shmgr pipeline info: segsize=%d\
                                              count=%d num_chunks=%d chunk_count_floor=%d\
                                              chunk_count_ceil=%d \n",
-                                             cellsize, (int) (count * type_size), num_chunks,
+                                             cellsize, (MPI_Aint) (count * type_size), num_chunks,
```

Comment:
I think this cast can probably be removed rather than changed to `(MPI_Aint)`.

Reply:
sure

Reply:
I second that. We should only use cast as last resort and remove them whenever possible. The compiler will not be able to warn anything with the cast.